### ISSUE TITLE
Hide default fields

### ISSUE THREAD

**Issue by willstott101** [django-helpdesk/django-helpdesk#503]
**Thanks for the awesome work here!** It would be great to see progress on the public facing web interface as well as the email system. I may do some work on this in the near future.

I noticed that there is currently no easy way to hide default fields... for our use case we wanted to hide the `due_date`, `priority`, and `assigned_to` fields from non-staff users. I ended up achieving that with the following patch. However some view code relies on the existence of the fields in the fields dictionary, so this is not complete as a PR.

Luckily the field expected by the view was the `assigned_to` field which has it's own special snowflake setting: `HELPDESK_CREATE_TICKET_HIDE_ASSIGNED_TO`

```
diff --git a/helpdesk/forms.py b/helpdesk/forms.py
index 7ab2024..d1b6994 100644
--- a/helpdesk/forms.py
+++ b/helpdesk/forms.py
@@ -317,6 +317,8 @@ class TicketForm(AbstractTicketForm):
         """
         super(TicketForm, self).__init__(*args, **kwargs)
         self._add_form_custom_fields()
+        for f in getattr(settings, 'EA_HIDDEN_TICKET_FIELDS', []):
+            self.fields.pop(f, None)
 
     def save(self, user=None):
         """
@@ -369,6 +371,8 @@ class PublicTicketForm(AbstractTicketForm):
         """
         super(PublicTicketForm, self).__init__(*args, **kwargs)
         self._add_form_custom_fields(False)
+        for f in getattr(settings, 'EA_PUBLIC_HIDDEN_TICKET_FIELDS', []):
+            self.fields.pop(f, None)
```

**Comment by gwasser** [comment]
Thanks! I've been looking at the email system a bit myself but as far as I'm aware no one is working on the public interface yet, so greatly appreciate your help with patches!

Offhand I'm not sure I like the introduction of a new `EA_HIDDEN_TICKET_FIELDS` variable. Do you think we might be able to accomplish it by making a more basic form class, and then making the staff form a subclass with more features, rather than popping off stuff?

**Comment by willstott101** [comment]
Yes that's indeed a possible technique. However that wouldn't really make it much more customisable. More recent versions on Django have a 'disabled' option for form fields. And there's also the HiddenField form Widget. I wonder if there's a way to implement this similar to how I've done it as a layer after building the class. But possibly at the template / widget level. Because:

1. View/Form code is expecting some of the fields to be present (refactor-able).

2. Forms may be a good place to define default data different to the defaults on the model (depending on who the form is intended for).

I agree that that setting isn't a pretty way to do it. One other technique might be a django admin style, subclass and register pattern? Where the helpdesk implementation provides all functionality. And sublcasses can override the hidden / visible options of the form widgets, rather than the form fields? Those subclasses can then register themselves and have the helpdesk code look for them. Maybe that's over-complicating it.

Simple not rendering forms in the template would also work. However exactly what that looks like (in terms of how the template is provided the fields that SHOULD be rendered), I'm not sure.

**Comment by arnuschky** [comment]
I just added PR #579 that adds three new settings that allow the operator to pre-define the queue, priority and due_date fields for public tickets. If one of these settings are present the corresponding input
field is hidden from the form.

It's simple but solves this problem for me in a relatively clean way.

**Comment by gwasser** [comment]
Thanks @arnuschky ! Would it be easy to add `assigned_to` as well to close out this ticket? We also need to update the documentation so users know about the new settings.

**Comment by arnuschky** [comment]
Should be fairly easy. The `assigned_to` setting didn't show up in my case, so I didn't add that. So basically we would set either a default owner or `None`, I assume?

Regarding the documentation, I can have a stab at documenting the settings.

**Comment by arnuschky** [comment]
> Would it be easy to add assigned_to as well to close out this ticket?

So the field `assigned_to` is not defined for the public form, and therefore there is no need to hide it. What did you have in mind?

**Comment by gwasser** [comment]
@arnuschky oh thanks for following up! This ticket mentioned `assigned_to` but I suppose that's not on the public form. I'll leave it open to remind that we should change the logged in form too. Thanks for your work on the public form!

**Comment by arnuschky** [comment]
Ok. I added documentation of the new settings to the PR.

**Comment by timthelion** [comment]
You can already hide most fields (except the one for attachments with a combination of setting the `_hide_fields_` and default values in the query parameters of the URL. We just need to find a better way of exposing this for non-iframe users.

https://github.com/django-helpdesk/django-helpdesk/blob/2b6ad7a2cffaf36d30a8414e6864a886bc135668/helpdesk/views/public.py#L102

See: https://github.com/django-helpdesk/django-helpdesk/blob/0.5.0/docs/integration.rst
